baseplate/core.py

```diff
@@ -151,6 +161,9 @@ def make_context(self, token):
 class AuthenticationContext(object):
     """Wrapper for the contextual authentication information
 
+    In general, this object should not be used directly but will rather be
+    wrapped by an :py:class:`EdgeRequestContext` object.
```

Comment:
I think so, I like that the logic around the authentication token is contained in it's own object rather than making `EdgeRequestContext` into an uber-object or splitting the logic between the different classes that care about authentication.